srtcore/api.cpp

```diff
@@ -1731,7 +1734,7 @@ void CUDTUnited::updateMux(
             && (i->second.m_iIpV6Only == s->m_pUDT->m_iIpV6Only)
             &&  i->second.m_bReusable)
          {
-            if (i->second.m_iPort == port)
+            if (i->second.m_iPort == port && CIPAddress::ipcmp(ip,i->second.m_piIP,i->second.m_iIPversion))
```

Comment:
If you want to add support of having port number with specific IP distinct and therefore allowed to use, you must make sure first that it also doesn't allow it to be combined with `INADDR_ANY`. In particular, this should be the behavior in case when the port number was found among the existing multiplexers:

1. If the found muxer is bound to `INADDR_ANY` (0.0.0.0), then:
1.1. If the new socket is also bound to `INADDR_ANY`, it should work the old way (no need to compare addresses)
1.2. If the new socket is bound to a specific IP address, the binding should fail:
     - simply do: `throw CUDTException(MJ_NOTSUP, MN_INVAL, 0);`
     - please update API documentation for `srt_bind()` that `SRT_EINVPARAM` error may be also thrown in such case
2. If the found muxer is bound to a specific IP address, then:
2.1. If the new socket is bound to `INADDR_ANY`, report failure as in 1.2.
2.2. If the new socket is bound to a specific IP address:
     - If it is the same address, allow to reuse this muxer
     - If the address is different, CONTINUE SEARCHING (merely the same as the current form)

(I personally think it might be easier to get the condition of where the new socket is to be bound,
that is, differently than I have described above, but with this it could be easily done by just making
two separate loops for these two cases, as they will contain different sets of conditions).